Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+                    @{Expression={$_.OsType};Label="OsType";Width=25;Alignment="left"},
+                    @{Expression={$_.OrchestrationMode};Label="OrchestrationMode";Width=25;Alignment="left"},
+                    @{Expression={$_.isExtPresent};Label="isExtPresent";Width=25;Alignment="left"},
+                    @{Expression={$_.isAADExtRolledback};Label="isAADExtRolledback";Width=25;Alignment="left"}
+     
+    if ($($VMSSRolledBack | Measure-Object).Count -gt 0 -or $($VMSSSkipped | Measure-Object).Count -gt 0)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "Rollback Summary: " -ForegroundColor $([Constants]::MessageType.Info)
+        
+        if ($($VMSSRolledBack | Measure-Object).Count -gt 0)
+        {
+            Write-Host "AAD extension is rolled back successfully on following Virtual Machine Scale Set(s) in the Subscription: " -ForegroundColor $([Constants]::MessageType.Update)
+            $VMSSRolledBack | Format-Table -Property $colsPropertyRollBack -Wrap
+
+            # Write this to a file.
+            $VMSSRolledBackFile = "$($backupFolderPath)\RolledBackVMSS.csv"
+            $VMSSRolledBack | Export-CSV -Path $VMSSRolledBackFile -NoTypeInformation
+            Write-Host "This information has been saved to" -NoNewline
+            Write-Host " [$($VMSSRolledBackFile)]" -ForegroundColor $([Constants]::MessageType.Update) 
+            Write-Host $([Constants]::SingleDashLine)
+        }
+
+        if ($($VMSSSkipped | Measure-Object).Count -gt 0)
+        {
+            Write-Host "Error installing AAD Extension on following Virtual Machine Scale Set(s) in the Subscription: " -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            $VMSSSkipped | Format-Table -Property $colsProperty -Wrap
+            
+            # Write this to a file.
+            $VMSSSkippedFile = "$($backupFolderPath)\RollbackSkippedVMSS.csv"
+            $VMSSSkipped | Export-CSV -Path $VMSSSkippedFile -NoTypeInformation
+            Write-Host "This information has been saved to" -NoNewline
+            Write-Host " [$($VMSSSkippedFile)]" -ForegroundColor $([Constants]::MessageType.Update)  
+            Write-Host $([Constants]::SingleDashLine)  
+        }
+    }
+}
+
+class VMSSExtensionList
+{
+[PSObject] GetVMSSExtensionList([string] $subscriptionId,[string] $ResourceGroupName,[string] $ResourceName)
+    {
+        $content = $null
+        try
+        {
+            #GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/extensions?api-version=2023-03-01
+            $armUri = "https://management.azure.com/subscriptions/$($subscriptionId)/resourceGroups/$($ResourceGroupName)/providers/Microsoft.Compute/virtualMachineScaleSets/$($ResourceName)/extensions?api-version=2023-03-01"
```

Comment:
we don;t have powershell command for this?

Reply:
Yes Tarun, we have PowerShell command which is giving extension details, extension's provisioning state is not available there this is why we have used Rest API.